perceval/backends/core/twitter.py

```diff
@@ -0,0 +1,371 @@
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import json
+import logging
+
+from grimoirelab.toolkit.datetime import datetime_utcnow, str_to_datetime
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient, RateLimitHandler
+from ...errors import BackendError
+
+CATEGORY_TWEET = "tweet"
+MAX_SEARCH_QUERY = 500
+
+TWITTER_URL = 'https://twitter.com/'
+TWITTER_API_URL = 'https://api.twitter.com/1.1/search/tweets.json'
+MAX_ITEMS = 100
+
+# Range before sleeping until rate limit reset
+MIN_RATE_LIMIT = 1
+
+# Time to avoid too many request exception
+SLEEP_TIME = 30
+
+TWEET_TYPE_MIXED = "mixed"
+TWEET_TYPE_RECENT = "recent"
+TWEET_TYPE_POPULAR = "popular"
+
+RATE_LIMIT_HEADER = "x-rate-limit-remaining"
+RATE_LIMIT_RESET_HEADER = "x-rate-limit-reset"
+
+logger = logging.getLogger(__name__)
+
+
+class Twitter(Backend):
+    """Twitter backend.
+
+    This class allows to fetch samples of tweets containing specific
+    keywords. Initialize this class passing API key needed
+    for authentication with the parameter `api_key`.
+
+    :param query: query to fetch tweets
+    :param api_token: token or key needed to use the API
+    :param max_items: maximum number of issues requested on the same query
+    :param sleep_for_rate: sleep until rate limit is reset
+    :param min_rate_to_sleep: minimun rate needed to sleep until
+         it will be reset
+    :param sleep_time: minimun waiting time to avoid too many request
+         exception
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_TWEET]
+
+    def __init__(self, query, api_token, max_items=MAX_ITEMS,
+                 sleep_for_rate=False, min_rate_to_sleep=MIN_RATE_LIMIT,
+                 sleep_time=SLEEP_TIME,
+                 tag=None, archive=None):
+        origin = TWITTER_URL
+
+        if len(query) >= MAX_SEARCH_QUERY:
+            msg = "Search query length exceeded %s, max is %s" % (len(query), MAX_SEARCH_QUERY)
+            raise BackendError(cause=msg)
+
+        super().__init__(origin, tag=tag, archive=archive)
+        self.query = query
+        self.api_token = api_token
+        self.max_items = max_items
+        self.sleep_for_rate = sleep_for_rate
+        self.min_rate_to_sleep = min_rate_to_sleep
+        self.sleep_time = sleep_time
+
+        self.client = None
+
+    def fetch(self, category=CATEGORY_TWEET, since_id=None, max_id=None,
+              geocode=None, lang=None,
+              include_entities=True, tweets_type=TWEET_TYPE_MIXED):
+        """Fetch the events from the server.
+
+        This method fetches tweets from the TwitterSearch API published in the last seven days.
+
+        :param category: the category of items to fetch
+        :param since_id: if not null, it returns results with an ID greater than the specified ID
```

Comment:
Instead of **if not null** I would say either **when it's set** or **if not `None`**